### DISCUSSION TITLE
custom vars not working with GetAtt - unrecognized property 'Fn::GetAtt'

### DISCUSSION THREAD

**Discussion by guyschlider** [serverless/serverless#10648] [Q&A] [answered]
Hi,

I want to support referring to DynamoDB tables resources but also for some stages to support providing ARNs for stream for already existing deployments of the service

I thought that using a custom variable with Fn::GetAtt only if not provided ARN - but when I try to point to a custom Fn:GetAtt I get this error:

```
Serverless: Configuration warning:
Serverless: at 'functions.test.events[0].stream': unrecognized property 'Fn::GetAtt'
```

**Comment by pgrzesik** [comment, marked as answer]
Hello @guyschlider - please provide more context and an example that is failing for you - from the provided information it's really hard to reason about what is failing. What version of the Framework are you using?

**Reply by guyschlider** [reply]
Hey, Thanks for the quick response!

Actually right now things are working for me, trying to understand what I've missed as I tried so many things. I will update here with a reproducible serverless.yml when I get to it.

**Reply by guyschlider** [reply]
Ok, so I found out
`- stream: ${self:custom.tableArn}` - NOT Working
but:
```
    - stream:
        type: dynamodb
        arn: ${self:custom.tableArn}
```

Does work
when tableArn is defined as:
```
custom:
    tableArn:
        Fn::GetAtt:
            - myTable
            - StreamArn
```

**Reply by pgrzesik** [reply]
How did you define it previously?

**Reply by guyschlider** [reply]
- stream: ${self:custom.tableArn}

Was working when tableArn was just a string of an ARN of the stream we were using

Now I'm migrating to use custom resources (so we don't have to create them manually) - I changed the custom.tableArn to use GetAtt instead. and when using 
`- stream: ${self:custom.tableArn}` I got:

```
Serverless: Configuration warning:
Serverless:   at 'functions.func.events[0].stream': unrecognized property 'Fn::GetAtt'
Serverless:   at 'functions.func.events[0].stream': unsupported configuration format
Serverless:  
Serverless: Learn more about configuration validation here: http://slss.io/configuration-validation
```

**Reply by pgrzesik** [reply]
Hello @guyschlider - this shorthand syntax of `- stream: <value>` is supported only if `value` is a string ARN or a name. If you want to use CF intrinsic functions such as `Fn::GetAtt`, you need to explicitly specify the type of stream and arn as in the other example you've shown:

```
- stream:
        type: dynamodb
        arn: ${self:custom.tableArn}
```

Otherwise it's not possible to know what type of stream that is, `kinesis` or `dynamodb`.

**Reply by guyschlider** [reply]
Got it. Thanks for the clarification.

**Reply by pgrzesik** [reply]
Happy I could help 🙌
